lib/ABI/FunctionType/Conversion.cpp

```diff
@@ -409,120 +409,123 @@ TCC::tryConvertingStackArguments(model::TypePath StackArgumentTypes,
     return llvm::SmallVector<model::Argument, 8>{};
   }
 
-  // Verify the alignment of the first argument.
-  if (Stack.Fields().empty()) {
-    revng_log(Log, "Stack struct has no fields.");
-  } else {
-    uint64_t FirstAlignment = *ABI.alignment(Stack.Fields().begin()->Type());
-    revng_assert(llvm::isPowerOf2_64(FirstAlignment));
-  }
-
-  // Go over all the `[CurrentArgument, TheNextOne]` field pairs.
-  // `CurrentRange` is used to keep track of the "remaining" ones.
-  // This allows us to keep its value when the loop to be aborted (which
-  // happens when we meet an argument we cannot just convert "as is").
   llvm::SmallVector<model::Argument, 8> Result;
   uint64_t InitialIndexOffset = Distributor.ArgumentIndex;
-  auto CurrentRange = std::ranges::subrange(Stack.Fields().begin(),
-                                            Stack.Fields().end());
-  bool NaiveConversionFailed = CurrentRange.empty();
-  while (CurrentRange.size() > 1) {
-    auto [CurrentArgument, TheNextOne] = takeAsTuple<2>(CurrentRange);
-
-    uint64_t NextAlignment = *ABI.alignment(TheNextOne.Type());
-    revng_assert(llvm::isPowerOf2_64(NextAlignment));
-
-    if (!canBeNext(Distributor,
-                   CurrentArgument.Type(),
-                   CurrentArgument.Offset(),
-                   TheNextOne.Offset(),
-                   NextAlignment)) {
-      // We met a argument we cannot just "add" as is.
-      NaiveConversionFailed = true;
-      break;
+  if (!model::hasMetadata(Stack)) {
```

Comment:
Add a comment about the ratio behind this

Reply:
Wanting to not drop types with metadata (if I remember correctly, now it's just custom names and comments) sounds pretty logical to me.

The rationale behind this change is slightly more pragmatic though: tests were failing if we were to unroll some of the structs. So ensuring they have a custom name was an easy and non-intrusive way to preserve them.

TL;DR: if converter ever meets a struct with name or comment or whatever new metadata we add in the future, it ensures it stays as a struct no matter what.

Reply:
I guess we should do a recursive check.

Anyway, add a comment.

Reply:
> I guess we should do a recursive check.

No need, we only ever unroll the `StackArgumentsType` struct itself. Everything nested any deeper is only ever treated as a struct.

In practice this only ever affects functions with a single struct stack argument, when the entire `StackArgumentsType` is replaced with the said struct.

Reply:
OK, resolve once you added a comment

Reply:
Done:
```cpp
    // NOTE: Only proceed with trying to "split" the stack struct up if it has
    //       no metadata attached: as in, it doesn't have a user-defined
    //       `CustomName` or a `Comment` or any other fields we'll mark as
    //       metadata in the future.
```